app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.github.sdp_begreen.begreen
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.matcher.IntentMatchers.*
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CompletableFuture
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class DatabaseActivityTest {
+    @get:Rule
+    val activityRule = ActivityScenarioRule(DatabaseActivity::class.java)
+
+    /*
+    Implementation of a local database
+     */
+    class LocalDataBase : Database() {
+        private val map: HashMap<String, String> = HashMap()
+        override fun get(key: String): CompletableFuture<String> {
+            return CompletableFuture.completedFuture(map[key])
+        }
+
+        override fun set(key: String, value: String) {
+            map[key] = value
+        }
+    }
+
+    @Before
+    fun before() {
+        // Change the used database to the local database
+        Database.db = LocalDataBase()
+    }
+
+    @Test
+    fun emailWrittenCorrectly() {
+        onView(withId(R.id.emailDB))
```

Comment:
I think the name of emailDB could be rename as it is an id of a text box

Reply:
I renamed all of the Buttons and EditTexts to follow the convention set in the main activity, thanks for pointing out that inconsistency!